docs/how-to/code/internal-user/rockcraft.yaml

```diff
@@ -0,0 +1,32 @@
+name: internal-user
+base: bare
+build-base: ubuntu@24.04
+version: 'latest'
+summary: Add an internal user inside a rock
+description: Add an internal user inside a rock, and run a service with it.
+platforms:
+    amd64:
+
+# [docs:rock-services]
+services:
+    whoami:
+        override: replace
+        startup: enabled
+        command: bash -c "whoami; sleep 5"
+        user: ubuntu
+        on-success: ignore 
+# [docs:rock-services-end]
+
+# [docs:rock-parts]
+parts:
+    my-part:
+        plugin: nil
+        stage-packages:
+            - base-passwd_data
+            - base-files_base
+            - bash_bins
+            - coreutils_bins
+        override-build: |
+            groupadd --root ${CRAFT_PART_INSTALL} ubuntu
+            useradd -d /home/ubuntu -s /bin/bash --root ${CRAFT_PART_INSTALL} -g ubuntu ubuntu
```

Comment:
hummm, I have my doubts about this.

Can you please try to run this with an ubuntu base? What happens? Can you print the resulting rock's `/etc/passwd`?

Having said that, I'd probably recommend using a different, nonconflicting, username

Reply:
your doubts are about the ubuntu user? it doesn't come with the ubuntu base by default. But I'm changing it to something else anyway.

Reply:
It does

```bash
$ docker run --rm ubuntu:noble cat /etc/passwd | grep ubuntu
ubuntu:x:1000:1000:Ubuntu:/home/ubuntu:/bin/bash
```